src/qbe_ir_generator.cpp

```diff
@@ -180,7 +186,30 @@ void QbeIrGenerator::Visit(const WhileStmtNode& while_stmt) {
   output << end_label << std::endl;
 }
 
-void QbeIrGenerator::Visit(const ForStmtNode& for_stmt) {}
+void QbeIrGenerator::Visit(const ForStmtNode& for_stmt) {
+  int label_num = NextLabelNum();
+  std::string pred_label = PrefixLabel("pred", label_num);
+  std::string body_label = PrefixLabel("loop_body", label_num);
+  std::string end_label = PrefixLabel("end", label_num);
+
+  // A for statement's loop initialization is the first clause to execute,
+  // whereas a for statement's predicate specifies evaluation made before each
+  // iteration. A step is an operation that is performed after each iteration.
+  // Skip predicate generation if it is a null expression.
+  for_stmt.loop_init->Accept(*this);
```

Comment:
```suggestion
  output << "# loop init" << std::endl;
  for_stmt.loop_init->Accept(*this);
```
How about adding a comment to indicate that the loop initialization expression starts from this point? :smile:
Other parts of the _for statement_ already have labels, so they are fine.